crates/cheatnet/src/cheatcodes/warp.rs

```diff
@@ -2,12 +2,56 @@ use crate::CheatnetState;
 use cairo_felt::Felt252;
 use starknet_api::core::ContractAddress;
 
+// Specifies which contracts to warp
+#[derive(Debug)]
+pub enum StartWarpTarget {
+    All(Felt252),
+    One((ContractAddress, Felt252)),
+    Multiple(Vec<(ContractAddress, Felt252)>),
+}
+
+// Specifies which contracts to stop warping
+#[derive(Debug)]
+pub enum StopWarpTarget {
+    All,
+    One(ContractAddress),
+    Multiple(Vec<ContractAddress>),
+}
```

Comment:
It's really similar to what i thought, but 
"Target" should be the 1st argument 
and 2nd argument of warp should be the target timestamp

And the "Target" enum would have associated addresses values, so that multiple would have an Vec of `ContractAddress`es, `One` would have a singular `ContractAddress` and `All` would be without extra arguments. 

Also as for the struct, I think that it should be named something more generic and placed higher up in the file hierarchy.

The changes i mentioned should be applied both to `snforge_std` and the rust code